File: Data/CodeExamples/SearchActions.vb
Imports System
Imports DevExpress.Spreadsheet
Imports DevExpress.Utils
Imports System.Drawing
Imports System.Globalization
Imports System.Collections.Generic


Namespace SpreadsheetExamples
	Public NotInheritable Class SearchActions
		Private Sub New()
		End Sub
		Private Shared Sub SimpleSearch(ByVal workbook As IWorkbook)
'			#Region "#SimpleSearch"
			Dim worksheet As Worksheet = workbook.Worksheets("SearchSample")
			workbook.Worksheets.ActiveWorksheet = worksheet
			worksheet.Visible = True

			' Find and highlight cells, containing text "the".
			Dim foundCells As IEnumerable(Of Cell) = worksheet.Search("the")
			For Each cell As Cell In foundCells
				cell.Fill.BackgroundColor = Color.Yellow
			Next cell
'			#End Region ' #SimpleSearch
		End Sub

		Private Shared Sub SearchWithOptions(ByVal workbook As IWorkbook)
'			#Region "#SearchWithOptions"
			Dim worksheet As Worksheet = workbook.Worksheets("SearchSample")
			workbook.Worksheets.ActiveWorksheet = worksheet
			worksheet.Visible = True

			' Find and highlight cells, containing case-sensitive text "The".
			Dim options As New SearchOptions()
			options.MatchCase = True
			Dim foundCells As IEnumerable(Of Cell) = worksheet.Search("The", options)
			For Each cell As Cell In foundCells
				cell.Fill.BackgroundColor = Color.Yellow
			Next cell
'			#End Region ' #SearchWithOptions
		End Sub
	End Class
End Namespace

File: Data/CodeExamples/SearchActions.cs
using System;
using DevExpress.Spreadsheet;
using DevExpress.Utils;
using System.Drawing;
using System.Globalization;
using System.Collections.Generic;


namespace SpreadsheetExamples {
    public static class SearchActions {
        static void SimpleSearch(IWorkbook workbook) {
            #region #SimpleSearch
            Worksheet worksheet = workbook.Worksheets["SearchSample"];
            workbook.Worksheets.ActiveWorksheet = worksheet;
            worksheet.Visible = true;

            // Find and highlight cells, containing text "the".
            IEnumerable<Cell> foundCells = worksheet.Search("the");
            foreach (Cell cell in foundCells)
                cell.Fill.BackgroundColor = Color.Yellow;
            #endregion #SimpleSearch
        }

        static void SearchWithOptions(IWorkbook workbook) {
            #region #SearchWithOptions
            Worksheet worksheet = workbook.Worksheets["SearchSample"];
            workbook.Worksheets.ActiveWorksheet = worksheet;
            worksheet.Visible = true;

            // Find and highlight cells, containing case-sensitive text "The".
            SearchOptions options = new SearchOptions();
            options.MatchCase = true;
            IEnumerable<Cell> foundCells = worksheet.Search("The", options);
            foreach (Cell cell in foundCells)
                cell.Fill.BackgroundColor = Color.Yellow;
            #endregion #SearchWithOptions
        }
    }
}
